add information how DelimitedFiles can be used (#2773)
JuliaData/DataFrames.jl: docs/src/man/importing_and_exporting.md

# Importing and Exporting Data (I/O)

> ## CSV files

Reviewer: ```suggestion
## CSV Files
```

                                                        135 rows omitted
```

Observe that in our example:
* `header` is a `Matrix` therefore we had to pass `vec(header)` to the `DataFrame`
  constructor;
* we broadcasted the `identity` function over the `iris_raw` data frame to perform
  narrowing of `eltype` of columns of `iris_raw`; the reason is that read in by
  the `readdlm` function is stored into a `data` `Matrix` so all columns in
  `iris_raw` initially have the same `eltype` -- in this case it had to be `Any`
  as some of the columns are numeric and some are string.

All such operations (and many more) are automatically handled by CSV.jl.

Similarly, you can use the `writedlm` function from the `DelimitedFiles` module to
save a data frame like this:

```julia
writedlm("test.csv", Iterators.flatten(([names(iris)], eachrow(iris))), ',')
```

As you can see the code required to transform `iris` into a proper input to the
`writedlm` function so that you can create the CSV file having the expected
format is not easy. Therefore CSV.jl is the preferred package to write CSV files
for data stored in data frames.

## Other formats

> Other, selected, data formats are supported for reading and writing in the
> following packages:

Reviewer: ```suggestion
Other data formats are supported for reading and writing in the following packages
(non exhaustive list):
```

> * Apache Arrow: [Arrow.jl](https://github.com/JuliaData/Arrow.jl)

Reviewer: ```suggestion
* Apache Arrow (including Feather v2): [Arrow.jl](https://github.com/JuliaData/Arrow.jl)
* Apache Feather (v1): [Feather.jl](https://github.com/JuliaData/Feather.jl)
```

* Apache Avro: [Avro.jl](https://github.com/JuliaData/Avro.jl)
* JSON: [JSONTables.jl](https://github.com/JuliaData/JSONTables.jl)
* Parquet: [Parquet.jl](https://github.com/JuliaIO/Parquet.jl)
* Stata, SPSS, and SAS: [StatFiles.jl](https://github.com/queryverse/StatFiles.jl)

> * XLSX: [XLSX.jl](https://github.com/felipenoris/XLSX.jl)

Reviewer: ```suggestion
* Microsoft Excel (XLSX): [XLSX.jl](https://github.com/felipenoris/XLSX.jl)
```

For reading and writing tabular data in Apache Arrow format use

> [Arrow.jl](https://github.com/JuliaData/Arrow.jl)

Author: OK.

> [Arrow.jl](https://github.com/JuliaData/Arrow.jl)

Author: added

> [Arrow.jl](https://github.com/JuliaData/Arrow.jl)

Author: while waiting for the approval I have added one more tutorial to the list of curated tutorials

> [Arrow.jl](https://github.com/JuliaData/Arrow.jl)

Author: @nalimilan - looks good now?

   8 │         5.0         3.4          1.5         0.2  Iris-setosa
  ⋮  │      ⋮           ⋮            ⋮           ⋮             ⋮
 144 │         6.8         3.2          5.9         2.3  Iris-virginica
 145 │         6.7         3.3          5.7         2.5  Iris-virginica
 146 │         6.7         3.0          5.2         2.3  Iris-virginica
 147 │         6.3         2.5          5.0         1.9  Iris-virginica
 148 │         6.5         3.0          5.2         2.0  Iris-virginica
 149 │         6.2         3.4          5.4         2.3  Iris-virginica
 150 │         5.9         3.0          5.1         1.8  Iris-virginica
                                                        135 rows omitted
```

Observe that in our example:
* `header` is a `Matrix` therefore we had to pass `vec(header)` to `DataFrame`
  constructor;
* we broadcasted `identity` function over the `iris_raw` data frame to perform
  narrowing of `eltype` of columns of `iris_raw`; the reason is that read in by
  the `readdlm` function is stored into a `data` `Matrix` so all columns in
  `iris_raw` initially have the same `eltype` -- in this case it had to be `Any`

>   as some of the folumns are numeric and some are string.

Reviewer: ```suggestion
  as some of the columns are numeric and some are string.
```
